### ISSUE TITLE
System hangs after multiples queries

### ISSUE THREAD

**Issue by sirspock** [KnowledgeCaptureAndDiscovery/wings#88]
System hangs after multiples queries.
I'm obtaining all the components from MINT and inserting on WINGS. However, the system froze.

## Expected Behavior
The system doesn't hang

## Current Behavior
The system hangs and wings data section does not upload. 

## Possible Solution
Related to #87?

## Steps to Reproduce

1. Clone the repo https://github.com/KnowledgeCaptureAndDiscovery/wings-api-scripts
2. Execute the mint.py. 

## Detailed Description

I sent 10-20 queries in 10 seconds and WINGS froze. 
The code that I'm running is available at https://github.com/KnowledgeCaptureAndDiscovery/wings-api-scripts/blob/master/mint.py

Also, the CPU usage is too high

![screen shot 2019-02-11 at 8 31 55 pm](https://user-images.githubusercontent.com/2365990/52603459-a8bb9e80-2e45-11e9-8aae-669bd806483e.png)

**Comment by IKCAP** [comment]
Could you tell me which is the resource that you are passing to mint.py ?

**Comment by sirspock** [comment]
This happens after all resources are loaded.
```
for i in $(cat resources.txt); do  python mint.py -r $i; sleep 10; done
```

**Comment by IKCAP** [comment]
Got it, thanks.

**Comment by sirspock** [comment]
I ran perf. perf top, which tells you which functions are in use on your computer right now

![screen shot 2019-02-12 at 9 05 03 am](https://user-images.githubusercontent.com/2365990/52634176-4a260d00-2ea5-11e9-8b0a-6706c48d1c5a.png)

**Comment by varunratnakar** [comment]
Couldn't replicate in a controlled environment:
1. Restarted WINGS just before the test
2. Create a new domain to populate
3. Did not use WINGS while the script was executing

I suspect that if I don't stick to 3 above, then I might encounter the locks. Will try that next.

Note: found a couple of bad names for inputs (Spaces in the labels. They should be replaced with underscores):
... /components/library.owl#DSSAT_cfg_simple_DSSAT weather file
... /components/library.owl#topoflow_cfg_simple_Rain rate txt file

**Comment by sirspock** [comment]
Were you able to read the data on the domain?

**Comment by varunratnakar** [comment]
Yes am able to read the data fine. Fixed the underscore bug, and ran again:
1. Restarted WINGS just before the test
2. Created a new domain to populate
3. *DID* use WINGS while the script was executing (read only, no writes)

Still cannot replicate.

**Comment by sirspock** [comment]
Ok. I'm going to test without using WINGS in my browser.

**Comment by varunratnakar** [comment]
Latest test:
1. Did *not* restart WINGS
2. Did *not* create a new domain
3. *Used* WINGS while the script was executing (read only)

Worked fine.

Note: I'm using the WINGS installation on my local machine.

**Comment by sirspock** [comment]
Can you test with this resource?

```
python mint.py -r https://w3id.org/mint/instance/cycles_outputs
```

**Comment by sirspock** [comment]
The other resources do not generate the issue

**Comment by varunratnakar** [comment]
Hmm, I still cannot reproduce this on a wings installation locally. Tried with clean and dirty domain both
